cmd/whereabouts_test.go

```diff
@@ -531,6 +531,176 @@ var _ = Describe("Whereabouts operations", func() {
+			"ipam": {
+			  "type": "whereabouts",
+			  "log_file" : "/tmp/whereabouts.log",
+              "log_level" : "debug",
+			  %s,
+        "ipRanges": [{
+			    "range": "192.168.10.1/24"
+        }]
+			}
+		  }`, backend)
+
+		args := &skel.CmdArgs{
+			ContainerID: "dummy",
+			Netns:       nspath,
+			IfName:      ifname,
+			StdinData:   []byte(conf),
+			Args:        cniArgs(podNamespace, podName),
+		}
+
+		ipamConf, cniVersion, err := config.LoadIPAMConfig([]byte(conf), cniArgs(podNamespace, podName), "")
+		Expect(err).NotTo(HaveOccurred())
+		k8sClient = newK8sIPAM(args.ContainerID, ipamConf, fakek8sclient.NewSimpleClientset(), fake.NewSimpleClientset(ipPool(ipamConf.Range, podNamespace)))
+
+		// Allocate the IP
+		r, raw, err := testutils.CmdAddWithArgs(args, func() error {
+			return cmdAdd(args, k8sClient, cniVersion)
+		})
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(strings.Index(string(raw), "\"version\":")).Should(BeNumerically(">", 0))
+
+		result, err := current.GetResult(r)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Gomega is cranky about slices with different caps
+
+		Expect(result.IPs[0].Address).To(Equal(mustCIDR("192.168.10.1/24")))
+
+		// Release the IP
+		err = testutils.CmdDelWithArgs(args, func() error {
+			return cmdDel(args, k8sClient)
+		})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("allocates DualStack address using IPRanges notation", func() {
+		const ifname string = "eth0"
+		const nspath string = "/some/where"
+
+		backend := fmt.Sprintf(`"kubernetes": {"kubeconfig": "%s"}`, kubeConfigPath)
+		conf := fmt.Sprintf(`{
+			"cniVersion": "0.3.1",
+			"name": "mynet",
+			"type": "ipvlan",
+			"master": "foo0",
+			"ipam": {
+			  "type": "whereabouts",
+			  "log_file" : "/tmp/whereabouts.log",
+              "log_level" : "debug",
+			  %s,
+        "ipRanges": [{
+			    "range": "192.168.10.1/24"
+        }, {
+          "range": "abcd::1/64"
+        }]
+			}
+		  }`, backend)
+
+		args := &skel.CmdArgs{
+			ContainerID: "dummy",
+			Netns:       nspath,
+			IfName:      ifname,
+			StdinData:   []byte(conf),
+			Args:        cniArgs(podNamespace, podName),
+		}
+
+		ipamConf, cniVersion, err := config.LoadIPAMConfig([]byte(conf), cniArgs(podNamespace, podName), "")
+		Expect(err).NotTo(HaveOccurred())
+		k8sClient = newK8sIPAM(args.ContainerID, ipamConf, fakek8sclient.NewSimpleClientset(), fake.NewSimpleClientset(ipPool(ipamConf.Range, podNamespace)))
+
+		// Allocate the IP
+		r, raw, err := testutils.CmdAddWithArgs(args, func() error {
+			return cmdAdd(args, k8sClient, cniVersion)
+		})
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(strings.Index(string(raw), "\"version\":")).Should(BeNumerically(">", 0))
+
+		result, err := current.GetResult(r)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Gomega is cranky about slices with different caps
+
+		Expect(result.IPs[0].Address).To(Equal(mustCIDR("192.168.10.1/24")))
+		Expect(result.IPs[1].Address).To(Equal(mustCIDR("abcd::1/64")))
```

Comment:
> Should we assert the length of `result.IPs` before assuming the indexes ?

Done :+1: 



> Or better yet:
> 
> ```go
> Expect(result.IPs).To(
>     ConsistOf(
>         mustCIDR("192.168.10.1/24"), 
>         mustCIDR("abcd::1/64")))
> ```

We can't use this because `results.IPs` is of type `[]*current.IPConfig` whereas `mustCIDR` if of type `net.IPNet`.
Additionally, I made a new function `ipConfigCIDR` which is of type `current.IPConfig`.
But this can't be used with `results.IPs` without creating a separate matcher. So I just asserted the lendth

Reply:
Fair enough. I'll sort that out once I can spare the time. 

Thanks.